COMMERCE-9973 - REST API POST /options throws an Exception when option values are populated

victorcdp opened the pull request:
Related issue: [COMMERCE-9973](https://issues.liferay.com/browse/COMMERCE-9973)

Problem was that there was no default value for the contextUriInfo, so it was always null.
Proposed fix is to add a default value so that a NPException doesn't happen.

brianikim commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest(master)/5289/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-5.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5289/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:relevant - 20 out of 20 jobs passed in 1 hour 13 minutes 41 seconds 255 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/f29d4f91891411eac91c4cad0d19b40402786ff8">f29d4f91891411eac91c4cad0d19b40402786ff8</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/020f12be3b7f68e4dcb3356fbd50440b139473fd">020f12be3b7f68e4dcb3356fbd50440b139473fd</a></p>20 out of 20 jobs PASSED<details><summary><strong>20 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest(master)/5289/">test-portal-acceptance-pullrequest(master)</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197161/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197160/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288724/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/325322/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/325325/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197163/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/325324/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/325319/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197159/">test-portal-acceptance-pullrequest-batch(master)/js-unit-private-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/325323/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197157/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197162/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197164/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/325321/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197156/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/325320/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288725/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/288726/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/197158/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-5.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5289/jenkins-report.html">here</a>.</h5></details></html>

brianikim commented:
See: https://github.com/SamZiemer/liferay-portal/pull/664

Commit:
LPS-169826 Return null instead of blank string

ccorreagg committed:
LPS-173905 Add reference in Portlet as the onAfterSave method of the listener is not called until the configuration is saved for the second time if configurationPolicy = REQUIRE is set

Commit:
LRQA-75616 Add a test that asserts Account Modals can be customized in style

julschong committed:
LPS-172618 Remove Stream usages from UserReindexerPerformanceOfLargeUserGroupInManySitesTest

julschong committed:
LPS-172618 Cleanup unused fields and change _getDocument return type to void since it is never used

julschong committed:
LPS-172618 Remove Stream usages from UserGroupCascadeReindexUsersTest

Commit:
LPS-169826 Set listTypeDefinitionId to ObjectField when importing an Object Definition

julschong committed:
LPS-172618 Remove Stream usages from UserIndexerIndexedFieldsByAssociationTest, use List.toString() and List.sort() instead of Stream

JanisZhang committed:
LPS-171132 Remove Optional usages in DataDefinitionContentTypeRegistry

JanisZhang committed:
LPS-171132 Remove Optional usages in DataDefinitionResourceImpl

claraizquierdo committed:
LPS-173245 Fix paddings and scroll in the properties sidebar

JanisZhang committed:
LPS-171132 Remove Stream usages in DataDefinitionResourceImpl

JanisZhang committed:
LPS-171132 Remove Stream usages in DataDefinitionDDMFormUtil

JanisZhang committed:
LPS-171132 Remove Stream usages in DataLayoutUtil

JanisZhang committed:
LPS-171132 Remove Stream usages in DataStorageUtil

sandrodw3 committed:
LPS-173488 Add react component for web content usages

sandrodw3 committed:
LPS-173488 buildLang

Luizcarlosqueiroz committed:
LPS-171616 Add view alert to add relationship entry

Luizcarlosqueiroz committed:
LPS-171616 Reuse menuitem macro

julschong committed:
LPS-172618 SF, simplified by inlining _getUserIds() and removing redundant sb operations

david-truong committed:
LPS-173700 cache transformed sql in publications also

balazssk committed:
LPS-172113 Should be false, not select-more-than-one

balazssk committed:
LPS-172113 Follow true-false pattern from JSP

balazssk committed:
LPS-17211 Revert existing new values to the original

drewbrokke committed:
LPS-173854 headless-admin-user-impl: adds DuplicateOrganizationExceptionMapper

Commit:
LPS-172991 Add checkRepetitions function to handle the visibility of the remove-duplicate-field

timpak committed:
LPS-173862 Split the test batches to account for unit and integration tests

timpak committed:
LPS-173862 Remove unused test batch

timpak committed:
LPS-173862 Include central requirements and semantic versioning

timpak committed:
LPS-173755 Create object environments test suite

ruben-pulido committed:
LPS-173580 Create LayoutClassedModelUsagesHelper

ruben-pulido committed:
LPS-173580 Make LayoutClassedModelUsagesHelper available through ServletContextUtil

baharturk committed:
COMMERCE-10657 Add three new functional tests

adolfopa committed:
LPS-70835 Check if a page with same title already exists

After sending a page to trash, it is possible to create a page with
the same title.  As titles must be unique per group, we need to check
for this case before trying to sync using LARs.

Note that staging is not affected by this -- it is not possible to
edit live -- but site templates use LARs to propagate its changes to
sites.

eduardo-diniz committed:
LPS-173464 Create Component Mobile

EugenioAntunes committed:
LPS-173785 Create Raylife Data Objects - Sales Goals Inserting Data

mbowerman committed:
LPS-173945 For writing content, we should use writeObject since the
content could be quite long and writeUTF can only handle content up to
65535 characters

mbowerman committed:
LPS-173945 Also need to use readObject now to read it back

mbowerman committed:
LPS-173945 Update tests

mbowerman committed:
LPS-173945 Add integration test

ruben-pulido committed:
LPS-173580 Create GetLayoutClassedModelUsagesStrutsAction

ruben-pulido committed:
LPS-173488 Don't use portletNamespace

ruben-pulido committed:
LPS-173580 Use LayoutClassedModelUsagesHelper

ruben-pulido committed:
LPS-173580 Add getUsagesData to LayoutClassedModelUsagesDisplayContext

melodywu committed:
LPS-173979 Add step to make case stable

dantewang committed:
LPS-171132 SF, rename

dantewang committed:
LPS-171132 SF, avoid checking the same condition repeatedly

Commit:
LPS-172991 Fix test: Wrap the PageProvider with FormProvider

ambrinchaudhary committed:
LPS-173880 Separate result's component

ambrinchaudhary committed:
LPS-173880 Show clay empty message

ambrinchaudhary committed:
LPS-173880 buildLang

Commit:
LPS-174017 Fix usage getActionVariables

ambrinchaudhary committed:
LPS-173881 Use a ClayList to show the results

david-truong committed:
LPS-173700 fix tests

Commit:
LPS-174017 Fix integration test

rafaellajordao committed:
LPS-166766 test fix for ContentPerformance#ChangeLanguageInDocumentDisplayPage

rafaellajordao committed:
LPS-166766 test fix for ContentPerformance#ChangeLanguageInWidgetPage

rafaellajordao committed:
LPS-166766 test fix for ContentPerformance#MetricsIconVisibleInBlogDisplayPage

rafaellajordao committed:
LPS-152127 test fix for ContentPerformance#MetricsIconVisibleInContentPage

rafaellajordao committed:
LPS-152127 test fix for ContentPerformance#MetricsIconVisibleInDocumentDisplayPage

rafaellajordao committed:
LPS-152127 test fix for ContentPerformance#MetricsIconVisibleInWidgetPage

Commit:
LPS-172368 Adjust

Commit:
LRAC-12840 Create property to build AC with matching pair for Asah 3.6.x

davisaints committed:
LPS-173939 Fix CanSetPaginationNumeric test

ambrinchaudhary committed:
LPS-173881 Highlight search query in results

veroglez committed:
LPS-169531 Remove unused scss

veroglez committed:
LPS-169531 Adapt css classes

ling-alan-huang committed:
LPS-174058 Auto SF

samanzanedo committed:
LPS-174138 Check if there is some JSON data before trying to normalize it

ambrinchaudhary committed:
LPS-172546 Animate as the Clay component

Commit:
LPS-172991 Fix: Add check to make sure there is a fieldName

ealonso committed:
LPS-172368 SF

ealonso committed:
LPS-172368 Rename

ealonso committed:
LPS-172368 Match with _getAssetPriority

Commit:
LPS-174163 Add ApproveEventWithInvitedUser to quaratine

murilostodolni committed:
LPS-174111 Add feature flag to LPS-155659

Commit:
LPS-174168 Add ViewProcessItemDetailPage to quarantine

Commit:
LPS-174178 Add CalendarUpgrade tests to quarantine

abhnerramos committed:
LPS-174030 Fix CanValidateSystemObjectWithExpressionBuilder test

sergiojimcos committed:
LPS-155809 else condition is missing

brianchandotcom committed:
LPS-171132 SF

brianchandotcom committed:
LPS-173700 SF

brianchandotcom committed:
LPS-172618 Rename

brianchandotcom committed:
LPS-70835 Wordsmith

Luizcarlosqueiroz committed:
LPS-171616 Add macros to allow page to be reloaded before proceding

JerryAdrianoo committed:
LPS-172299 Fix ObjectRelationships#CanRelateOneToManyCustomObjectEntries

brianchandotcom committed:
LPS-174111 Move package

brianchandotcom committed:
LPS-173880 Wordsmith

brianchandotcom committed:
LPS-173880 Regen

brianikim committed:
COMMERCE-10078 Add keyword support for SKU and Names in Order Search

brianchandotcom committed:
LPS-173580 & LPS-173488 SF

brianchandotcom committed:
COMMERCE-10078 Rename

pat270 committed:
LPS-168603 Add addresses cancel link should be announced as button in screen readers

JessieZhao0714 committed:
LRQA-79315 Add a custom property

timpak committed:
LPS-173755 Don't run functional tests on other environments until we stabilize them

Commit:
LPS-174335 Fix TITLE_FIELD path and macro

Commit:
LPS-174350 talend-runtime - fix compile time error

thaiscabrals committed:
LPS-156132 Automate FormsSubmissions#CannotNavigateThroughThePagesOnPublicPagesWithContainsValidationError

samuelleolopes committed:
LPS-174226 Created Redirection for New Task

samuelleolopes committed:
LPS-174331 Updated Upload Parameters

joshchong committed:
LPS-174376 Remove txtai container at the end of each run

patriciajeanperez committed:
LRQA-68075 Rename tests

patriciajeanperez committed:
LRQA-68075 Sort

patriciajeanperez committed:
LRQA-68075 Add description to tests

Commit:
COMMERCE-9973 Add default value for contextUriInfo

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

victorcdp commented:
ci:test:sf

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/faf682e989d8872bdea67d3b17c2d4318c888502">faf682e989d8872bdea67d3b17c2d4318c888502</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/victorcdp/liferay-portal/tree/COMMERCE-9973">COMMERCE-9973</a><br/>Branch GIT ID: <a href="https://github.com/victorcdp/liferay-portal/commit/969b990c8e9fa1839ca61ba8bb5bc70fad145d73">969b990c8e9fa1839ca61ba8bb5bc70fad145d73</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-11.liferay.com/job/test-portal-source-format/6727/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-11.liferay.com/userContent/jobs/test-portal-source-format/builds/6727/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-11.liferay.com/job/test-portal-source-format/6727/">test-portal-source-format#6727</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-11.liferay.com/userContent/jobs/test-portal-source-format/builds/6727/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/brianikim/liferay-portal/pull/55">brianikim#55</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2184208146">[master] ci:test:sf - victorcdp &gt; brianikim - PR#55 - 2023-02-02[09:34:44]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-11.liferay.com/job/publish-testray-report/3494/">publish-testray-report#3494</a><br/></div></div>
